core/include/usBundle.h

```diff
@@ -47,143 +48,297 @@ typedef ServiceReference<void> ServiceReferenceU;
+     * the bundle has successfully started and moves to the {@code STATE_ACTIVE}
+     * state.
+     *
+     * If the bundle has a {@link Constants#ACTIVATION_LAZY lazy activation
+     * policy}, then the bundle may remain in this state for some time until the
+     * activation is triggered.
+     *
+     * The value of {@code STATE_STARTING} is 0x00000008.
+     */
+    STATE_STARTING = 0x00000008,
+
+    /**
+     * The bundle is in the process of stopping.
+     *
+     * A bundle is in the {@code STATE_STOPPING} state when its {@link #Stop(int)
+     * Stop} method is active. A bundle is in this state when the bundle's
+     * {@link BundleActivator#Stop(const BundleContext&)} method
+     * is called. When the {@code BundleActivator#Stop} method completes the bundle
+     * is stopped and moves to the {@code STATE_RESOLVED} state.
+     *
+     * The value of {@code STATE_STOPPING} is 0x00000010.
+     */
+    STATE_STOPPING = 0x00000010,
+
+    /**
+     * The bundle is now running.
+     *
+     * A bundle is in the {@code STATE_ACTIVE} state when it has been successfully
+     * started and activated.
+     *
+     * The value of {@code STATE_ACTIVE} is 0x00000020.
+     */
+    STATE_ACTIVE = 0x00000020
+  };
+
+  enum StartOptions : uint32_t {
+
+    /**
+     * The bundle start operation is transient and the persistent autostart
+     * setting of the bundle is not modified.
+     *
+     * This bit may be set when calling {@link #Start(uint32_t)} to notify the
+     * framework that the autostart setting of the bundle must not be modified.
+     * If this bit is not set, then the autostart setting of the bundle is
+     * modified.
+     *
+     * @see #Start(uint32_t)
+     */
+    START_TRANSIENT = 0x00000001,
+
+    /**
+     * The bundle start operation must activate the bundle according to the
+     * bundle's declared
+     * {@link Constants#PLUGIN_ACTIVATIONPOLICY activation policy}.
+     *
+     * This bit may be set when calling {@link #Start(uint32_t)} to notify the
+     * framework that the bundle must be activated using the bundle's declared
+     * activation policy.
+     *
+     * @see Constants#PLUGIN_ACTIVATIONPOLICY
+     * @see #Start(uint32_t)
+     */
+    START_ACTIVATION_POLICY = 0x00000002
+  };
+
+  enum StopOptions {
+
+    /**
+     * The bundle stop is transient and the persistent autostart setting of the
+     * bundle is not modified.
+     *
+     * This bit may be set when calling {@link #Stop(uint32_t)} to notify the
+     * framework that the autostart setting of the bundle must not be modified.
+     * If this bit is not set, then the autostart setting of the bundle is
+     * modified.
+     *
+     * @see #Stop(uint32_t)
+     */
+    STOP_TRANSIENT = 0x00000001
+  };
+
+  // Bundle(const Bundle &) = default;
+  // Bundle(Bundle&&) = default;
+  // Bundle& operator=(const Bundle&) = default;
+  // Bundle& operator=(Bundle&&) = default;
```

Comment:
If I'm reading the rules for[ implicitly generated move constructors](http://en.cppreference.com/w/cpp/language/move_constructor) correctly, the compiler will not generate a move constructor by default in this case because there is a user-defined destructor. I dumped the symbols from a CppMicroServices.dll built with VS 2015 and noticed there is no move constructor. If I re-build without a user defined destructor, I do see a compiler generated move constructor.

Since the intent is to have a default move constructor (which I agree is the right thing to do), either the destructor needs to be removed or the move/copy constructors need to be user defined (to support VS 2013).